src/sidebar/components/search-status-bar.js

```diff
@@ -24,113 +22,161 @@ const countVisibleAnns = annThread => {
+     * A search (filter) query, visible to the user in the search bar, is
+     * currently applied
+     */
+    filtered: !!filterQuery,
+    /**
+     * @type {Boolean}
+     * The client has a currently-applied focus on a single user. Superseded by
+     * `filtered` mode.
+     */
+    focused: focusModeFocused && !filterQuery,
+    /**
+     * @type {Boolean}
+     * 0 - n annotations are currently "selected", by, e.g. clicking on highlighted
+     * text in the host page, direct-linking to an annotation, etc. Superseded by
+     * `filtered` mode.
+     */
+    selected: (() => {
+      if (directLinkedGroupFetchFailed) {
+        return true;
+      }
+      return (
+        !!selectionMap && Object.keys(selectionMap).length > 0 && !filterQuery
+      );
+    })(),
+  };
 
   const visibleCount = useMemo(() => {
     return countVisibleAnns(thread);
   }, [thread]);
 
-  const filterResults = (() => {
-    switch (visibleCount) {
-      case 0:
-        return `No results for "${storeState.filterQuery}"`;
-      case 1:
-        return '1 search result';
-      default:
-        return `${visibleCount} search results`;
-    }
-  })();
-
-  const focusResults = (() => {
-    switch (visibleCount) {
-      case 0:
-        return `No annotations for ${storeState.focusModeUserPrettyName}`;
-      case 1:
-        return 'Showing 1 annotation';
-      default:
-        return `Showing ${visibleCount} annotations`;
-    }
-  })();
+  // Each "mode" has corresponding descriptive text about the number of
+  // matching/applicable annotations and, sometimes, a way to clear the
+  // filter
+  const modeText = {
+    filtered: (() => {
+      switch (visibleCount) {
+        case 0:
+          return `No results for "${filterQuery}"`;
+        case 1:
+          return '1 search result';
+        default:
+          return `${visibleCount} search results`;
+      }
+    })(),
+    focused: (() => {
+      switch (visibleCount) {
+        case 0:
+          return `No annotations for ${focusModeUserPrettyName}`;
+        case 1:
+          return 'Showing 1 annotation';
+        default:
+          return `Showing ${visibleCount} annotations`;
+      }
+    })(),
+    selected: (() => {
+      // Generate the proper text to show on the clear-selection button.
+      // For non-user-focused modes, we can display the number of annotations
+      // that will be visible if the selection is cleared (`counts.annotations`)
+      // but this number is inaccurate/misleading when also focused on a user.
+      let selectedText;
+      switch (selectedTab) {
+        case uiConstants.TAB_ORPHANS:
+          selectedText = 'Show all annotations and notes';
+          break;
+        case uiConstants.TAB_NOTES:
+          selectedText = 'Show all notes';
+          if (counts.notes > 1 && !modes.focused) {
+            selectedText += ` (${counts.notes})`;
+          }
+          break;
+        case uiConstants.TAB_ANNOTATIONS:
+        default:
```

Comment:
nit:

my preference is not including a `default` if its invalid condition. e.g. can `selectedTab` ever not be one `TAB_ORPHANS`,  `TAB_NOTES` or `TAB_ANNOTATIONS` ?

Reply:
Technically, there's nothing to _stop_ `selectedTab` from being any value (I don't think?). I was trying to communicate that the `annotations` tab is the "most likely" active tab, but I can kill the `default`.